x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -6,64 +6,101 @@
+import { getDefaultSingleMetricViewerPanelTitle } from './get_default_panel_title';
 
 export interface SingleMetricViewerInitializerProps {
   bounds: TimeRangeBounds;
-  defaultTitle: string;
   initialInput?: Partial<SingleMetricViewerEmbeddableInput>;
-  job: MlJob;
-  onCreate: (props: Partial<SingleMetricViewerEmbeddableUserInput>) => void;
+  mlApiServices: MlApiServices;
+  onCreate: (props: SingleMetricViewerEmbeddableUserInput) => void;
   onCancel: () => void;
 }
 
 export const SingleMetricViewerInitializer: FC<SingleMetricViewerInitializerProps> = ({
-  defaultTitle,
   bounds,
   initialInput,
-  job,
   onCreate,
   onCancel,
+  mlApiServices,
 }) => {
-  const isNewJob = initialInput?.jobIds !== undefined && initialInput?.jobIds[0] !== job.job_id;
+  const isMounted = useMountedState();
+  const newJobUrl = useMlLink({ page: ML_PAGES.ANOMALY_DETECTION_CREATE_JOB });
+  const [jobIds, setJobIds] = useState(initialInput?.jobIds ?? []);
+  const titleManuallyChanged = useRef(!!initialInput?.title);
 
-  const [panelTitle, setPanelTitle] = useState<string>(defaultTitle);
+  const [job, setJob] = useState<MlJob | undefined>();
+  const [panelTitle, setPanelTitle] = useState<string>(initialInput?.title ?? '');
   const [functionDescription, setFunctionDescription] = useState<string | undefined>(
     initialInput?.functionDescription
   );
   // Reset detector index and entities if the job has changed
   const [selectedDetectorIndex, setSelectedDetectorIndex] = useState<number>(
-    !isNewJob && initialInput?.selectedDetectorIndex ? initialInput.selectedDetectorIndex : 0
+    initialInput?.selectedDetectorIndex ?? 0
   );
   const [selectedEntities, setSelectedEntities] = useState<MlEntity | undefined>(
-    !isNewJob && initialInput?.selectedEntities ? initialInput.selectedEntities : undefined
+    initialInput?.selectedEntities
   );
-
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
   const isPanelTitleValid = panelTitle.length > 0;
 
+  useEffect(
+    function setUpPanel() {
+      async function fetchJob() {
+        const { jobs } = await mlApiServices.getJobs({ jobId: jobIds.join(',') });
+
+        if (isMounted() && jobs.length === 1) {
+          setJob(jobs[0]);
+          setErrorMessage(undefined);
+        }
+      }
+
+      if (jobIds.length === 1) {
+        if (!titleManuallyChanged.current) {
+          setPanelTitle(getDefaultSingleMetricViewerPanelTitle(jobIds[0]));
+        }
+        // Fetch job if a jobId has been selected and if there is no corresponding fetched job or the job selection has changed
+        if (mlApiServices && jobIds.length === 1 && jobIds[0] !== job?.job_id) {
+          fetchJob().catch((error) => {
+            const errorMsg = extractErrorMessage(error);
+            setErrorMessage(errorMsg);
+          });
+        }
+      }
+    },
+    [isMounted, jobIds, mlApiServices, panelTitle, job?.job_id]
+  );
```

Comment:
If find this effect quite difficult to follow. First of all, the job selection in a single metric embeddable allows only one job. Why do we have an unnecessary complexity of tracking `jobIds`. It should be just `jobId`, which a primitive value that easier to keep track of for React. 

```suggestion
 const jobId = jobIds?.[0];

  useEffect(
    function setUpPanel() {
     if (!jobId) {
     // escape if job id is not selected
       return;
     }
    
      async function fetchJob() {
        const { jobs } = await mlApiServices.getJobs({ jobId: jobIds.join(',') });

        if (isMounted()) {
          setJob(jobs[0]);
          setErrorMessage(undefined);
        }
      }
      
       if (!titleManuallyChanged.current) {
          setPanelTitle(getDefaultSingleMetricViewerPanelTitle(jobIds[0]));
       }
        // Fetch job if a jobId has been selected and if there is no corresponding fetched job or the job selection has changed
        if (jobId !== job?.job_id) {
          fetchJob().catch((error) => {
            const errorMsg = extractErrorMessage(error);
            setErrorMessage(errorMsg);
          });
        }
    },
    [isMounted, jobId, mlApiServices, panelTitle, job?.job_id]
  );
```

Reply:
That's a great point - definitely better to keep it as a single job id here - updated in aedae88d210a9522a57bbc6a42334b73577150a7